### ISSUE TITLE
Setting the Host header does not set the request's host

### ISSUE THREAD

**Issue by sefaphlvn** [TwinProduction/gatus#60]
How can I send request to ip address with request header Host?  Is there any way?

Example:

```services:
  - name: test
    url: "https://5.5.5.5/asd"
    headers:
      Host: www.google.com
    interval: 5s
    conditions:
      - "[STATUS] == 200"
      - "[RESPONSE_TIME] < 1000"
```

**This scenario not working**

```<TITLE>Invalid URL</TITLE>
</HEAD><BODY>
<H1>Invalid URL</H1>
The requested URL "&#91;no&#32;URL&#93;", is invalid.<p>
Reference&#32;&#35;9&#46;b5df3a17&#46;1608149667&#46;2020b03
</BODY></HTML>```

**Comment by TwinProduction** [comment]
If you try to access the website using the given `url`, does it work?

The fact that it's returning an error means that technically, the request works. It's probably an issue with the reverse proxy if I had to guess.

Also, you might have to set `insecure` to `true`, because the certificate is bound to the domain, not the IP, and since you're using `https`, the certificate will likely return an error.

Here's an example with `example.org` (IP: `93.184.216.34`):
```yaml
services:
  - name: example
    url: "https://93.184.216.34/" 
    insecure: true
    headers:
      Host: example.org
    interval: 30s
    conditions:
      - "[STATUS] == 404"
      - "[IP] == 93.184.216.34"
      - "[RESPONSE_TIME] < 1000"
```
P.S. This specific example probably returns a 404 because they have a reverse proxy or a router that doesn't use the `Host` header.

**Comment by sefaphlvn** [comment]
Thanks for quick answer. When I try with postman its working.

but this doesn't work. Could you check yaml below?
```
services:
  - name: akamai
    url: "http://23.58.223.185/s/7/500/9749975957554.jpg"
    headers:
      Host: productimages.hepsiburada.net
    interval: 30s
    conditions:
      - "[STATUS] == 200"
      - "[IP] == 23.58.223.185"
      - "[RESPONSE_TIME] < 1000"
```

**Comment by TwinProduction** [comment]
It works as expected @sefaphlvn.
The fact that the response is a 400 indicates that the server is answering Gatus' request.

The reason why you're running into a problem is that, well, if you go on `http://23.58.223.185/s/7/500/9749975957554.jpg`, you get a 400.

Anyways, why don't you just use the URL `http://productimages.hepsiburada.net/s/7/500/9749975957554.jpg` instead?
```yaml
services:
  - name: akamai
    url: "http://productimages.hepsiburada.net/s/7/500/9749975957554.jpg"
    interval: 30s
    conditions:
      - "[STATUS] == 200"
      - "[RESPONSE_TIME] < 1000"
```

**Comment by sefaphlvn** [comment]
You are right. If request with ip address without host header like this `http://23.58.223.185/s/7/500/9749975957554.jpg` response is 400. But when I set a host header it's returns 200 on postman.

Why don't use just the URL because we have multi datacenter and we are using dns loadbalancer. I need to check spesific site from ip address.

I guess Gatus doesn't send the HOST header.

This is exactly what I want to do
```
services:
  - name: akamai
    url: "http://23.58.223.185/s/7/500/9749975957554.jpg"
    headers:
      Host: productimages.hepsiburada.net
    interval: 30s
    conditions:
      - "[STATUS] == 200"
      - "[RESPONSE_TIME] < 1000"
  - name: google
    url: "http://x.x.x.x/s/7/500/9749975957554.jpg"
    headers:
      Host: productimages.hepsiburada.net
    interval: 30s
    conditions:
      - "[STATUS] == 200"
      - "[RESPONSE_TIME] < 1000"     
```

**Comment by TwinProduction** [comment]
Sorry for the delay, just got off work.

I gave it a shot and you're completely right. Turns out that setting the `Host` header in isn't sufficient for `http.Request`, `http.Request.Host` also needs to be updated.

**Comment by TwinProduction** [comment]
Released in [v1.9.2](https://github.com/TwinProduction/gatus/releases/tag/v1.9.2)
